sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala

```diff
@@ -233,21 +244,7 @@ class V2SessionCatalog(catalog: SessionCatalog)
         throw QueryCompilationErrors.tableAlreadyExistsError(ident)
     }
 
-    val table = loadTable(ident)
-
-    // Check if the schema of the created table matches the given schema.
-    // TODO: move this check in loadTable to match the behavior with
-    // existing file data sources.
-    if (schema.nonEmpty) {
-      val tableSchema = CharVarcharUtils.replaceCharVarcharWithStringInSchema(
-        table.columns().asSchema)
-      if (!DataType.equalsIgnoreNullability(tableSchema, schema)) {
-        throw QueryCompilationErrors.dataSourceTableSchemaMismatchError(
-          table.columns().asSchema, schema)
-      }
-    }
-
-    table
+    null // Return null to save the `loadTable` call for CREATE TABLE without AS SELECT.
```

Comment:
Hmm, internally we can catch callers of this and add `Option` like this does. A concern is it might break some usages externally? The API return type is not changed, so users might need to dig into this change and find that this API could return null.

Reply:
Is this worth noting in migration guide?

Reply:
Hmm, `TableCatalog` is still `Evolving` API. So it might be okay then.

Reply:
This is a developer API that we expect users to implement, not to invoke. There is no way people can access an instance of `TableCatalog` unless they use private APIs (e.g. `SessionState#catalogManager`) or they instantiate their own `TableCatalog` implementations.